src/checkers/inference/model/ComparableConstraint.java

```diff
@@ -11,18 +13,38 @@
     private final Slot first;
     private final Slot second;
 
-    protected ComparableConstraint(Slot first, Slot second, AnnotationLocation location) {
+    private ComparableConstraint(Slot first, Slot second, AnnotationLocation location) {
         super(Arrays.asList(first, second), location);
         this.first = first;
         this.second = second;
     }
 
-    protected ComparableConstraint(Slot first, Slot second) {
+    private ComparableConstraint(Slot first, Slot second) {
         super(Arrays.asList(first, second));
         this.first = first;
         this.second = second;
     }
 
+    protected static Constraint create(Slot first, Slot second, AnnotationLocation location,
```

Comment:
Doesn't seem very nice that after calling `ComparableConstraint#create()` method, the topmost type `Constraint` is returned instead of `ComparableConstraint`

Reply:
We do normalize constant-constant pairs to T/F constraint